### PR TITLE
Add webhook template functionality to the console

### LINKED ISSUE #945: Tiles for supported webhook integations
#### Summary

Tiles for supported webhook integrations.

#### Why do we need this?

To show recognizable logos for users of upstream platforms and make configuration easier.

#### What is already there? What do you see now?

Manual webhook configuration, i.e. URL, headers, format.

#### What is missing? What do you want to see?

Tiles that provide URL, headers and format.

#### How do you propose to implement this?

1. AS file based configuration of webhook templates and API - @adriansmares 
   - id
   - name, description, logo, info url, documentation url
   - format to use
   - url with path variables (for app ID, device ID, DevEUI)
   - field for user security token that can be used in various ways (e.g. `Authorization` header with optional prefix). Also the name of the user security token (i.e. "Subscription ID", "API Key")
2. AS rpcs to serve webhook templates (part of `pkg/applicationserver/io/web`) - @adriansmares 
3. Console support for fetching templates, rendering tiles and asking user security token - @kschiffer 

We should keep the template ID used in the application webhook so that we can render the logo of the webhook. Apart from that, for editing and deleting the webhook, I would suggest using the existing edit/delete functionality, so that the template is only used for creating the webhook.

#### Can you do this yourself and submit a Pull Request?

Can review

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
Closes #945
References #1175 

This PR adds webhook template functionality to the console. Webhooks can then be created using webhook templates that have been set up in the stack config.

<details>
<summary>Screenshots</summary>

image
image
image
</details>

#### Changes
- Add `url-templates` package to assemble the base URLs
- Extend JS SDK with webhook template RPCs
- Add redux primitives and store logic for webhook templates
- Add webhook template form which takes the template and assembles the respective templated webhook form
- Update template form to show information about associated template
- Add webhook template info component
- Add template chooser view
- Extend and adapt webhook related routes and views

#### Notes for Reviewers
To set up the template(s), refer to @adriansmares [instructions](https://github.com/TheThingsNetwork/lorawan-stack/issues/945#issuecomment-584258732)
I've used the following template setup:
<details>
<summary>example.yml</summary>

```yml
template-id: example
name: Example Template
description: A simple webhook template
logo-url: https://upload.wikimedia.org/wikipedia/commons/3/30/Vector-based_example.svg
info-url: https://www.example.com
documentation-url: https://www.iana.org/domains/reserved
fields:
  - id: region
    name: Server region
    description: On which region should the service endpoint land on
    secret: false
    default-value: eu
format: json
headers:
  x-test: test {region}
create-downlink-api-key: true
base-url: https://www.example.com{/region}
paths:
  uplink-message: /uplink
```
</details>

I have implemented it so that the flow with choosing the template is only applied when there is at least one template configured. Otherwise, there will be no change in the UX.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by johanstokking** [comment]
Awesome. @adriansmares can you thoroughly review this?

@bafonins also the frontend technicalities.

**Comment by adriansmares** [comment]
I've just finished testing this one out and I love it. Good job @kschiffer ! UX for webhooks just goes through the roof when you no longer see 1000 options but instead only the relevant ones for the integration you're trying to add.

I believe we can go further with this: at the moment it does not leverage the fact that you can save the values of the fields as part of the webhook using the `template_fields` mapping in the webhook (see [webhook definition](https://github.com/TheThingsNetwork/lorawan-stack/blob/master/api/api.md#ttn.lorawan.v3.ApplicationWebhook)). If during creation the fields are saved, then editing the webhook can reuse the same form as creation - the user will be fully shielded from the full options of a webhook.

The drawback of this is that the user cannot tweak the insides of a webhook created by a template, so one possible solution is to have a button `Convert to custom webhook` which basically sets the `template_id` and `template_fields` to `null` and allows the user to use the webhook as if it were a classical webhook.

File: pkg/webui/console/components/webhook-template-form/index.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+    const webhook = {
+      ids: {
+        webhook_id: values.webhook_id,
+      },
+      template_ids: template.ids,
+      format: template.format,
+      headers,
+      base_url: urlTemplate.parse(template.base_url).expand(fields),
+      uplink_message: template.uplink_message,
```
Comment (adriansmares) [inline pkg/webui/console/components/webhook-template-form/index.js:70]: These should be get expanded as well.

**Comment by kschiffer** [comment]
> I believe we can go further with this: at the moment it does not leverage the fact that you can save the values of the fields as part of the webhook using the template_fields mapping in the webhook (see webhook definition). If during creation the fields are saved, then editing the webhook can reuse the same form as creation - the user will be fully shielded from the full options of a webhook.

Cool, let's do that. @johanstokking do you agree?

> The drawback of this is that the user cannot tweak the insides of a webhook created by a template, so one possible solution is to have a button Convert to custom webhook which basically sets the template_ids and template_fields to null and allows the user to use the webhook as if it were a classical webhook.

Good idea, will implement it this way.

**Comment by johanstokking** [comment]
Yes, but I would rather push this to Next Up and consider other priorities first at the moment. But in general, yes, I do support this completely.

**Pull request merged**